test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Models/PackageItemListViewModelTests.cs

```diff
@@ -37,5 +58,401 @@ public void LocalSources_PackageReader_NotNull()
+            };
+
+            BitmapSource result = GetFinalIconBitmap(packageItemListViewModel).Result;
+
+            VerifyImageResult(result);
+            Assert.Same(DefaultPackageIcon, result);
+        }
+
+        [Fact]
+        public void IconUrl_RelativeUri_ReturnsDefault()
+        {
+            // relative URIs are not supported in viewmodel.
+            var iconUrl = new Uri(@"resources/testpackageicon.png", UriKind.Relative);
+            var packageItemListViewModel = new PackageItemListViewModel()
+            {
+                IconUrl = iconUrl
+            };
+
+            BitmapSource result = GetFinalIconBitmap(packageItemListViewModel).Result;
+
+            VerifyImageResult(result);
+            Assert.Same(DefaultPackageIcon, result);
+        }
+
+        [Fact]
+        public void IconUrl_WithLocalPathAndColorProfile_LoadsImage()
+        {
+            var iconUrl = new Uri(@"resources/grayicc.png", UriKind.Relative);
+            var packageItemListViewModel = new PackageItemListViewModel()
+            {
+                IconUrl = iconUrl
+            };
+
+            BitmapSource result = GetFinalIconBitmap(packageItemListViewModel).Result;
+
+            VerifyImageResult(result);
+            Assert.NotSame(DefaultPackageIcon, result);
+        }
+
+        [Fact]
+        public void IconUrl_WithValidImageUrl_FailsDownloadsImage_ReturnsDefault()
+        {
+            var iconUrl = new Uri("http://fake.com/image.png");
+
+            var packageItemListViewModel = new PackageItemListViewModel()
+            {
+                IconUrl = iconUrl
+            };
+
+            BitmapSource result = GetFinalIconBitmap(packageItemListViewModel).Result;
+
+            VerifyImageResult(result);
+            Assert.Equal(DefaultPackageIcon, result);
+        }
+
+        [LocalOnlyTheory]
+        [InlineData("icon.png", "icon.png", "icon.png", "")]
+        [InlineData("folder/icon.png", "folder\\icon.png", "folder/icon.png", "folder")]
+        [InlineData("folder\\icon.png", "folder\\icon.png", "folder\\icon.png", "folder")]
+        public void IconUrl_EmbeddedIcon_HappyPath_LoadsImage(
+            string iconElement,
+            string iconFileLocation,
+            string fileSourceElement,
+            string fileTargetElement)
+        {
+            using (var testDir = TestDirectory.Create())
+            {
+                // Create decoy nuget package
+                var zipPath = Path.Combine(testDir.Path, "file.nupkg");
+                CreateDummyPackage(
+                    zipPath: zipPath,
+                    iconName: iconElement,
+                    iconFile: iconFileLocation,
+                    iconFileSourceElement: fileSourceElement,
+                    iconFileTargetElement: fileTargetElement,
+                    isRealImage: true);
+
+                // prepare test
+                UriBuilder builder = new UriBuilder(new Uri(zipPath, UriKind.Absolute))
```

Comment:
`var`

Reply:
yep fixed in next iteration ... all the uribuilders.